src/ord.rs

```diff
@@ -178,51 +159,26 @@ impl fmt::Display for OrdOps {
 ///
 /// This is created by the `predicate::{gt, ge, lt, le}` functions.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct OrdPredicate<T>
-where
-    T: fmt::Debug + PartialOrd,
-{
+pub struct OrdPredicate<T> {
     constant: T,
     op: OrdOps,
 }
 
-impl<T> Predicate<T> for OrdPredicate<T>
-where
-    T: fmt::Debug + PartialOrd,
-{
-    fn eval(&self, variable: &T) -> bool {
-        match self.op {
-            OrdOps::LessThan => variable.lt(&self.constant),
-            OrdOps::LessThanOrEqual => variable.le(&self.constant),
-            OrdOps::GreaterThanOrEqual => variable.ge(&self.constant),
-            OrdOps::GreaterThan => variable.gt(&self.constant),
-        }
-    }
-
-    fn find_case<'a>(&'a self, expected: bool, variable: &T) -> Option<reflection::Case<'a>> {
-        utils::default_find_case(self, expected, variable).map(|case| {
-            case.add_product(reflection::Product::new(
-                "var",
-                utils::DebugAdapter::new(variable).to_string(),
-            ))
-        })
-    }
-}
-
-impl<'a, T> Predicate<T> for OrdPredicate<&'a T>
+impl<T, P> Predicate<T> for OrdPredicate<P>
```

Comment:
What about the predicates outside of this file?

Reply:
I'm not aware of any other general purpose predicates that could benefit from the same approach, but if you have any suggestions then I can take a look. I would suggest that the specific use predicates of path and str are outside the scope of this PR though.

Reply:
#49 (linked from #20) touched predicates in `iter.rs` as well.